v2.2/architecture/life-of-a-distributed-transaction.md

```diff
@@ -0,0 +1,204 @@
+
+What we detail below is a simplified version of the CockroachDB transaction model. For more detail, check out [the transaction architecture documentation](transaction-layer.html).
+
+#### Resolving Write Intents
+
+If an operation encounters a write intent for a key, it attempts to "resolve" the write intent by checking the write intent's transaction's state. If the write's intent's transaction is...
+
+- `COMMITTED`, this operation converts the write intent to a regular key-value pair, and then proceeds as if it had read that value instead of a write intent.
+- `ABORTED`, this operation discards the write intent and reads the next-most-recent value from RocksDB.
+- `PENDING`, the new transaction attempts to "push" the write intent's transaction by moving that transaction's timestamp forward (i.e. ahead of this transaction's timestmap). The push succeeds only if this transaction's priority is explicitly higher; i.e. it has a `HIGH` priority or the transaction whose write intent it encountered has a `LOW` priority.
+  
+  If the push succeeds, the operation continues.
+
+  If this push fails, this transaction goes into the [`TxnWaitQueue`](https://www.cockroachlabs.com/docs/stable/architecture/transaction-layer.html#txnwaitqueue) on this node. Only once the blocking transaction completes (i.e., commits or aborts), the incoming transaction can continue.
+
+Check out our architecture documentation for more information about [CockroachDB's transactional model]((https://www.cockroachlabs.com/docs/stable/architecture/transaction-layer.html). 
+
+#### Read Operations
+
+If the read doesn't encounter a write intent and the key-value operation is meant to serve a read, it can simply use the value it read from the leaseholder's instance of RocksDB. This works because the leaseholder had to be part of the Raft consensus group for any writes to complete, meaning it must have the most up-to-date version of the range's data.
+
+The leaseholder aggregates all read responses into a `BatchResponse` that will get returned to the gateway node's `DistSender`.
+
+As we mentioned before, each read operation also updates the timestamp cache.
+
+### Write Operations
+
+After guaranteeing that there are no existing write intents for the keys, `BatchRequest`'s key-value operations are converted to [Raft operations](replication-layer.html#raft) and have their values converted into write intents.
+
+The leaseholder then proposes these Raft operations to the Raft group leader. Most often, the leaseholder and the Raft leader are the same node, but there are situations where the roles might drift to different nodes. However, when the two roles are not collocated on the same physical machine, CockroachDB will attempt to relocate them on the same node at the next opportunity.
+
+## Raft Leader
```

Comment:
I'd make a note that the Range leaseholder and that Range's Raft leader are almost always the same node.